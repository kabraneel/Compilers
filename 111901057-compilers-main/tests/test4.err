IR PRETTY PRINT
.globl main
main:
li t1, 3
li t2, 2
li t3, 70
label1:
beq t2, t3, label2
move t4, t1
move t9, t4
li t8, 1
syscall
addi t2, t2, 1
j label1
label2:
li t8, 10
syscall

 BASIC BLOCKS 
.globl main

main:
li $t1, 3
li $t2, 2
li $t3, 70

label1:
beq $t2, $t3, label2

move $t4, $t1
move $a0, $t4
li $v0, 1
syscall
addi $t2, $t2, 1
j label1


label2:
li $v0, 10
syscall

