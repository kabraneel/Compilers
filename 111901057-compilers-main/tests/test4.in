x := 3;
for i:= 2 to 70 
do
	print x;
done;
