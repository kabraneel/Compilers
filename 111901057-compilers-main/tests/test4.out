.globl main
main:
li $t1, 3
li $t2, 2
li $t3, 70
label1:
beq $t2, $t3, label2
move $t4, $t1
move $a0, $t4
li $v0, 1
syscall
addi $t2, $t2, 1
j label1
label2:
li $v0, 10
syscall
